### DISCUSSION TITLE
Is it possible to return several validation errors from `validate` function?

### DISCUSSION THREAD

**Discussion by vovkvlad** [react-hook-form/react-hook-form#10142] [Q&A]
Hi!
I couldn't find any relevant answers in examples or in discussion:
I wonder if it is possible to return several validation error messages for the same field at once, and not return them one-by-one? 
For example, I have a form with a password field and I want to show all rules that are not met for the current field value (several error messages rendered beneath the input, one per validation rule that is not met).

If I do it next way:
```js
// other register or <Controller /> props
validate: {
              hasSpecialSymbols: (v) =>
                ['_', '-', '~', '@'].some((symbol) => v.includes(symbol)),
              enoughLength: (v) => v.length > 5,
              hasUppercaseLetters: (v) => /[A-Z]/.test(v),
            },
```
Then both `useForm().formState.erorrs.password` and `fieldState.error` would show one validation error at a time.
If I try to do smth like this:
```js
validate: (v) => {
              const errorsArr = [];
              const hasSpecialSymbols = ['_', '-', '~', '@'].some((symbol) =>
                v.includes(symbol)
              );
              if (!hasSpecialSymbols) {
                errorsArr.push(
                  `Should contain one of '_', '-', '~', '@' symbols`
                );
              }

              if (v.length < 5) {
                errorsArr.push('Password must have at least 5 characters');
              }

              if (!/[A-Z]/.test(v)) {
                errorsArr.push(
                  'Password must contain at least one uppercase letter'
                );
              }

              return errorsArr;
            },
```
Then form's state will have only error with type `validate` for this field, and no error message. And there is no way getting access to returned array of error messages.

So my question is: Is it me missing something, or it's just not possible with React Hook Form?

**Comment by Sajarin-M** [comment]
You can use the first method in combination with the `criteriaMode: 'all'` option in `useForm` and you will have access to all errors at once in `formState.errors[yourFieldName].types`

```diff
const { formState } = useForm({
  defaultValues: {
    password: '',
  },
+ criteriaMode: 'all',
});

console.log(formState.errors.password?.types)
```

**Reply by vovkvlad** [reply]
Sorry I my question was not completely clear, but I', interested in returning all validation rules that are not passing for a single input. I am not interested in errors for all form fields. 

Already discussed it in this issue: https://github.com/react-hook-form/react-hook-form/issues/10145
It turns out it's not feasible at the moment
